<?xml version="1.0"?>
<launch>

<!-- Cart base  -->

<include file="$(find om_modbus_master)/launch/om_modbusRTU.launch"/>
<node pkg="om_cart" name="om_cart" type="om_cart">
</node>

<!-- Localization -->
<include file="$(find hdl_localization)/launch/hdl_localization_livox.launch" />

<!--PointCloud2 to Laser_scan -->
<node name="local_obstacle_filter" pkg="amr_ros" type="local_obstacle_filter">
    <param name="OBST_HIGHT_MIN_Z" value="0.0"/>
    <param name="OBST_HIGHT_MAX_Z" value="2.0"/>
</node>

<!--map load-->
<arg name="static_map" default="ikeda/ikeda_map.yaml"/>
<!--map server-->
<node pkg="map_server" type="map_server" name="map_server" args="$(find amr_ros)/maps/$(arg static_map)">
</node>

<node pkg="tf" type="static_transform_publisher" name="base_link_to_livox_frame" args="0.23 0 0 0 0 0 base_link livox_frame 10"/>

<node pkg="tf" type="static_transform_publisher" name="base_link_to_sonar_center" args="0 0.1 0.5 0 0 0 base_link sonar_frame_center 10"/>
<node pkg="tf" type="static_transform_publisher" name="base_link_to_sonar_left" args="0 0.3 0 0 0 0 base_link sonar_frame_left 10"/>
<node pkg="tf" type="static_transform_publisher" name="base_link_to_sonar_right" args="0 -0.3 0 0 0 0 base_link sonar_frame_right 10"/>


<!--move_base -->
<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find amr_ros)/config/navi_auto/move_base.yaml" command="load" />

    <rosparam file="$(find amr_ros)/config/navi_auto/costmap_common.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find amr_ros)/config/navi_auto/costmap_common.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find amr_ros)/config/navi_auto/global_costmap.yaml" command="load" />
    <rosparam file="$(find amr_ros)/config/navi_auto/local_costmap.yaml" command="load" />
    <rosparam file="$(find amr_ros)/config/navi_auto/global_planner.yaml" command="load" />
    <rosparam file="$(find amr_ros)/config/navi_auto/dwa_local_planner.yaml" command="load" />

    <remap from="cmd_vel" to="cmd_vel"/>
    <remap from="odom" to="odom"/>
    <remap from="map" to="map"/>

</node>

<node name="usb_cam_node" pkg="usb_cam" type="usb_cam_node" args="_video_device:=/dev/video2">

</node>


<!--Start 3DLidar -->
<include file="$(find livox_ros_driver2)/launch_ROS1/rviz_MID360.launch" />


<arg name="gui" default="true" />
<param name="use_gui" value="$(arg gui)"/>
<arg name="rvizconfig" value="$(find amr_ros)/rviz/navi_auto.rviz"/>
<node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />


</launch>
